test/tstDistributedKokkosToolsAnnotations.cpp

```diff
@@ -0,0 +1,226 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborX_EnableDeviceTypes.hpp" // ARBORX_DEVICE_TYPES
+#include <ArborX_DistributedSearchTree.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <string>
+
+#include "Search_UnitTestHelpers.hpp"
+
+#if (KOKKOS_VERSION >= 30200) // callback registriation from within the program
+                              // was added in Kokkkos v3.2
+
+BOOST_AUTO_TEST_SUITE(DistributedKokkosToolsAnnotations)
+
+namespace tt = boost::test_tools;
+
+bool isPrefixedWith(std::string const &s, std::string const &prefix)
+{
+  return s.find(prefix) == 0;
+}
+
+BOOST_AUTO_TEST_CASE(is_prefixed_with)
+{
+  BOOST_TEST(isPrefixedWith("ArborX::Whatever", "ArborX"));
+  BOOST_TEST(!isPrefixedWith("Nope", "ArborX"));
+  BOOST_TEST(!isPrefixedWith("Nope::ArborX", "ArborX"));
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(
+    distributed_search_tree_distributed_search_tree_allocations_prefixed,
+    DeviceType, ARBORX_DEVICE_TYPES)
+{
+  Kokkos::Tools::Experimental::set_allocate_data_callback(
+      [](Kokkos::Profiling::SpaceHandle handle, const char *label,
+         void const *ptr, uint64_t size) {
+        std::cout << label << '\n';
+        BOOST_TEST((isPrefixedWith(label, "ArborX::DistributedSearchTree::") ||
+                    isPrefixedWith(label, "ArborX::BVH::") ||
+                    isPrefixedWith(label, "ArborX::Sorting::") ||
+                    isPrefixedWith(label, "Testing::")));
+      });
+
+  { // empty
+    auto tree = makeDistributedSearchTree<DeviceType>(MPI_COMM_WORLD, {});
+  }
+
+  { // one leaf per process
+    auto tree = makeDistributedSearchTree<DeviceType>(
+        MPI_COMM_WORLD, {
+                            {{{0, 0, 0}}, {{1, 1, 1}}},
+                        });
+  }
+
+  { // two leaves per process
+    auto tree = makeDistributedSearchTree<DeviceType>(
+        MPI_COMM_WORLD, {
+                            {{{0, 0, 0}}, {{1, 1, 1}}},
+                            {{{0, 0, 0}}, {{1, 1, 1}}},
+                        });
+  }
```

Comment:
These corner case make sense on the BVH because they explore different code path.  I am not sure what they bring to the table for the distributed tree.

Reply:
I think it makes sense to have at least the empty case and something else but I am happy to drop the empty case as well if you are happier with that.

Reply:
Agreed. Leave empty and remove one and two leaves cases.